File: Presentation.Web/app/models/generic/NamedEntityDTO.ts
module Kitos.Models.Generic.NamedEntity {

    export interface NamedEntityDTO {
        id: number;
        name: string;
    }

    export interface NamedEntityWithEnabledStatusDTO extends NamedEntityDTO{
        disabled : boolean;
    }

    export interface NamedEntityWithDescriptionAndExpirationStatusDTO extends NamedEntityDTO {
        expired: boolean;
        description: string;
    }
    export interface NamedEntityWithUserFullNameDTO extends NamedEntityDTO {
        userFullName: string;
    }

    export interface EntityWithOrganizationRelationshipDTO extends NamedEntityDTO {
        organization : Organization.ShallowOrganizationDTO
    }
}

File: Presentation.Web/Models/API/V1/NamedEntityDTO.cs
namespace Presentation.Web.Models.API.V1
{
    public class NamedEntityDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public NamedEntityDTO(int id, string name)
        {
            Id = id;
            Name = name;
        }
    }
}